server/src/main/java/io/druid/client/DirectDruidClient.java

```diff
@@ -315,6 +322,7 @@ public InputStream nextElement()
               nodeTimeMs,
               byteCount.get() / (0.001 * nodeTimeMs) // Floating math; division by zero will yield Inf, not exception
           );
+          QueryMetrics<? super Query<T>> queryMetrics = acquireQueryMetrics();
```

Comment:
How about just calling `acquireQueryMetrics()` -- the shadowing of a reference with another reference to the same object is weird.